src/Entity/Invoice.php

```diff
@@ -94,66 +95,173 @@ public function __construct(
         }
     }
 
-    public function getId(): string
+    /** @deprecated this function is deprecated and will be removed in v1.0 - Use Invoice::getId() instead */
+    public function id(): string
     {
-        return $this->id();
+        return $this->id;
     }
 
-    public function id(): string
+    public function getId()
     {
         return $this->id;
     }
 
+    public function setId($id): void
+    {
+        $this->id = $id;
+    }
+
+    /** @deprecated this function is deprecated and will be removed in v1.0 Use Invoice::getNumber() instead */
     public function number(): string
     {
         return $this->number;
     }
 
+    public function getNumber(): string
+    {
+        return $this->number;
+    }
+
+    public function setNumber(string $number): void
```

Comment:
`Invoice` is an immutable entity, we have a constructor for creating a proper instance, so we shouldn't add set methods